Dolphin scripts/Entrance Randomizer/lib/graph_creation.py

```diff
@@ -0,0 +1,136 @@
+from pathlib import Path
+
+from lib.constants import *  # noqa: F403
+from lib.constants import __version__
+from lib.transition_infos import Area
+from lib.types_ import SeedType
+
+STARTING_AREA_COLOR = "#ff8000"  # Orange
+UPGRADE_AREAS_COLOR = "#0080ff"  # Blue
+STORY_TRIGGER_AREAS_COLOR = "#ff0000"  # Red
+UPGRADE_AREAS = {
+    LevelCRC.PLANE_COCKPIT,
+    LevelCRC.BITTENBINDERS_CAMP,
+    LevelCRC.MOUTH_OF_INTI,
+    LevelCRC.FLOODED_COURTYARD,
+    LevelCRC.TURTLE_MONUMENT,
+    LevelCRC.NATIVE_VILLAGE,
+    LevelCRC.RENEGADE_HEADQUARTERS,
+    LevelCRC.CAVERN_LAKE,
+    LevelCRC.MOUNTAIN_SLED_RUN,
+    LevelCRC.MOUNTAIN_OVERLOOK,
+    LevelCRC.APU_ILLAPU_SHRINE,
+}
+IMPORTANT_STORY_TRIGGER_AREAS = {
+    LevelCRC.ALTAR_OF_AGES,
+    LevelCRC.ST_CLAIRE_DAY,
+    LevelCRC.ST_CLAIRE_NIGHT,
+    LevelCRC.GATES_OF_EL_DORADO,
+}
+
+
+def create_vertices(
+    transitions_map: Mapping[tuple[int, int], tuple[int, int]],
+    starting_area: int,
+):
+    output_text = ""
+    areas_randomized: list[Area] = []
+    for original, redirect in transitions_map.items():
+        if TRANSITION_INFOS_DICT[original[0]] not in areas_randomized:
+            areas_randomized.append(TRANSITION_INFOS_DICT[original[0]])
+        if TRANSITION_INFOS_DICT[redirect[1]] not in areas_randomized:
+            areas_randomized.append(TRANSITION_INFOS_DICT[redirect[1]])
+    counter_x = 0
+    counter_y = 0
+    for area in areas_randomized:
+        output_text += (
+            f'<node positionX="{counter_x * 100 + counter_y * 20}" '
+            + f'positionY="{counter_x * 50 + counter_y * 50}" '
+            + f'id="{area.small_id}" mainText="{area.name}" '
+        )
+        if area.area_id == starting_area:
+            output_text += (
+                'ownStyles = "{&quot;0&quot;:{&quot;fillStyle&quot;:&quot;"
+                + STARTING_AREA_COLOR
+                + "&quot;}}" '
+            )
+        elif area.area_id in UPGRADE_AREAS:
+            output_text += (
+                'ownStyles = "{&quot;0&quot;:{&quot;fillStyle&quot;:&quot;"
+                + UPGRADE_AREAS_COLOR
+                + "&quot;}}" '
+            )
+        elif area.area_id in IMPORTANT_STORY_TRIGGER_AREAS:
+            output_text += (
+                'ownStyles = "{&quot;0&quot;:{&quot;fillStyle&quot;:&quot;"
+                + IMPORTANT_STORY_TRIGGER_AREAS_COLOR
+                + "&quot;}}" '
+            )
+        output_text += "></node>\n"
+        row_length = 10
+        counter_x += 1
+        if counter_x == row_length:
+            counter_x = 0
+            counter_y += 1
+    return output_text
+
+
+def create_edges(transitions_map: Mapping[tuple[int, int], tuple[int, int]]):
+    output_text = ""
+    connections: list[tuple[int, int]] = []
+    connections_two_way: list[tuple[int, int]] = []
+    connections_one_way: list[tuple[int, int]] = []
+    for original, redirect in transitions_map.items():
+        connections.append((original[0], redirect[1]))
```

Comment:
This can be achieved with a list comprehension
```suggestion
    connections = [(original[0], redirect[1]) for original, redirect in transitions_map.items()]
    connections_two_way: list[tuple[int, int]] = []
    connections_one_way: list[tuple[int, int]] = []
```